internal/server/deviceflow.go

```diff
@@ -64,46 +64,105 @@ retry:
 }
 
 // GetDeviceFlowStatus is an API handler for checking the status of a device
-// flow login. The response status can be pending, rejected, expired, or confirmed.
+// flow login. The response status can be pending, expired, or confirmed.
 func (a *API) GetDeviceFlowStatus(c *gin.Context, req *api.DeviceFlowStatusRequest) (*api.DeviceFlowStatusResponse, error) {
 	rctx := getRequestContext(c)
-	dfar, err := access.FindDeviceFlowAuthRequest(rctx, req.DeviceCode)
+
+	dfar, err := data.GetDeviceFlowAuthRequest(rctx.DBTxn, data.GetDeviceFlowAuthRequestOptions{ByDeviceCode: req.DeviceCode})
 	if err != nil {
-		return nil, err
+		if errors.Is(err, internal.ErrNotFound) {
+			return nil, err
+		}
+		return nil, fmt.Errorf("%w: error retrieving device flow auth request: %v", internal.ErrUnauthorized, err)
 	}
 
 	if dfar.ExpiresAt.Before(time.Now()) {
 		return &api.DeviceFlowStatusResponse{
-			Status:     "expired",
+			Status:     api.DeviceFlowStatusExpired,
 			DeviceCode: dfar.DeviceCode,
 		}, nil
 	}
 
-	if dfar.AccessKey != nil {
+	if !dfar.Approved() {
 		return &api.DeviceFlowStatusResponse{
-			Status:     "confirmed",
+			Status:     api.DeviceFlowStatusPending,
 			DeviceCode: dfar.DeviceCode,
-			LoginResponse: &api.LoginResponse{
-				UserID:           dfar.AccessKey.IssuedFor,
-				Name:             dfar.AccessKey.IssuedForName,
-				AccessKey:        string(dfar.AccessKeyToken),
-				Expires:          api.Time(dfar.AccessKey.ExpiresAt),
-				OrganizationName: dfar.Organization.Name,
-			},
 		}, nil
 	}
 
+	user, err := data.GetIdentity(rctx.DBTxn, data.GetIdentityOptions{ByID: dfar.UserID})
+	if err != nil {
+		return nil, fmt.Errorf("%w: retrieving approval user: %v", internal.ErrUnauthorized, err)
+	}
+
+	// TODO : there's a lot of repeated logic in here and the Login() endpoint
+	// The Login() endpoint implemention needs to be re-worked into more atomic pieces that can
+	// be re-used here for setting cookies, creating an access key and a LoginResponse
+	accessKey := &models.AccessKey{
+		IssuedFor:     user.ID,
+		IssuedForName: user.Name,
+
+		// Share the same provider ID that was used to approve
+		ProviderID:          dfar.ProviderID,
+		ExpiresAt:           time.Now().UTC().Add(a.server.options.SessionDuration),
+		InactivityTimeout:   time.Now().UTC().Add(a.server.options.SessionInactivityTimeout),
+		InactivityExtension: a.server.options.SessionInactivityTimeout,
+		Scopes:              models.CommaSeparatedStrings{models.ScopeAllowCreateAccessKey},
```

Comment:
device keys shouldn't be allowed to create more keys

Reply:
That seems like it's going to be a problem. Previously we were able to login from the CLI and create access keys. If device flow doesn't count as a login, then there's no way for someone to truly login via the CLI.

Reply:
I've added a new scope for access keys that allows them to approve device flow requests. The issue in the PR was not that device flow keys could create access keys (they are equivalent to login keys), but that they can continue to "renew" themselves by approving subsequent device flow requests.